hs_core/views/__init__.py

```diff
@@ -208,6 +211,15 @@ def update_metadata_element(request, shortkey, element_name, element_id, *args,
 
     return HttpResponseRedirect(request.META['HTTP_REFERER'])
 
+@api_view(['GET'])
+def file_download_url_mapper(request, shortkey, filename):
+    """ maps the file URIs in resourcemap document to django_irods download view function"""
+
+    authorize(request, shortkey, view=True, edit=True, full=True, superuser=True)
```

Comment:
@hyi The download url mapper view function and any rest api view class/function that does a redirect to the irods download function will fail if the decorator is removed from the download view function.